lib/sinon/util/core/is-es-property.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+var deprecated = require("@sinonjs/commons").deprecated;
+
+/**
+ * Verify an object's property is not an ECMAScript Module property
+ *
+ * As the exports from a module is immutable we cannot alter the exports
+ * using spies or stubs. Let the consumer know this to avoid bug reports
+ * on weird error messages.
+ *
+ * @param {Object} object The object to examine
+ * @param {string} property The property to examine
+ */
+module.exports = function(object, property) {
+    // eslint-disable-next-line no-underscore-dangle
+    if (!object || !object.__esModule) {
+        return;
+    }
+
+    var own = Object.getOwnPropertyDescriptor(object, property);
+    if (!own.get || own.set) {
+        return;
+    }
+
+    var msg =
+        'Property "' +
+        property +
+        '" cannot be overriden, it appears to be a ES Module read-only property. ' +
+        "These can be generated with re-exports in Typescript.";
```

Comment:
Instead of all of this, I would just drop the esModule thing, check if the prop had a getter and setter (like you do) and if these were configurable. If no on the latter, we could go into the switch block. Otherwise, if _for some reason_ Babel in the future supported exporting configurable props you could still modify the props using Sinon's getter/setter functionality.